haskell/private/actions/repl.bzl

```diff
@@ -1,5 +1,6 @@
 """GHCi REPL support"""
 
+load(":private/context.bzl", "render_env")
```

Comment:
Wouldn't it be possible to factor all this file out with the new `haskell_repl` rule?

Reply:
Not quite at the moment. `haskell_repl` is its own rule and can return a `DefaultInto` and thereby properly use runfiles. `haskell/private/actions/repl.bzl` creates an extra output in the regular `haskell_*` rules and cannot return a dedicated `DefaultInfo` with runfiles. Instead it has to use the `ln` trick to force it's dependencies and refer to them in `bazel-out`.

I agree that there is duplication that we don't want to keep around in the long term. One possible way forward could be to drop `haskell/private/actions/repl.bzl` completely and replace it by the `haskell/repl.bzl` aspect, similar to `haskell/haddock.bzl`. But, I'd prefer to do that in a separate PR.